### PR TITLE
[KEP-2400] Add `pod_swap_usage_bytes` as an expected metric in resource metric e2e test

### LINKED ISSUE #119400: Test failure: [sig-node] ResourceMetricsAPI [NodeFeature:ResourceMetrics] when querying /resource/metrics should report resource usage through the resource metrics api
### Which jobs are failing?

Node e2e features is failing after PR for graduating swap to beta is merged. 

### Which tests are failing?

https://testgrid.k8s.io/sig-node-cri-o#ci-crio-cgroupv1-node-e2e-features

### Since when has it been failing?

After https://github.com/kubernetes/kubernetes/pull/118865 is merged. 

### Testgrid link

https://testgrid.k8s.io/sig-node-cri-o#ci-crio-cgroupv1-node-e2e-features

### Reason for failure (if possible)

Test needs to be adjusted for the extra field added for swap usage.

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig node

### PR DESCRIPTION
**Pull request by iholder101**
#### What type of PR is this?

/kind cleanup
/kind failing-test

#### What this PR does / why we need it:
Recently, swap stats [were added](https://github.com/kubernetes/kubernetes/pull/118865) to `/stats/summary` and `/metrics/resource`.
These changes caused the `ResourceMetricsAPI` e2e test to fail at periodic jobs [1], that do not expect the `pod_swap_usage_bytes` metric.

This PR adds expectation for `pod_swap_usage_bytes` metric to the test, which will now cause it to pass.

[1] https://testgrid.k8s.io/sig-node-cri-o#ci-crio-cgroupv1-node-e2e-features

#### Which issue(s) this PR fixes:

Fixes #119400

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by iholder101** [comment]
@pacoxu @ffromani

Tests pass and PR is ready to review.

**Changes summary:**
The `pod_swap_usage_bytes` metric is not always reported, as it depends if swap is enabled on the node. Therefore, I've added the `IgnoreMissing` option to `MatchKeys`, this way even if `pod_swap_usage_bytes` is missing the test would pass.

However, I wanted to keep ensuring that all of the other metrics are always reported. Therefore, I've explicitly ensured they were reported by using `And(..., haveKeys(...))`.

File: test/e2e_node/resource_metrics_test.go [hunk 1]
```diff
@@ -165,3 +165,18 @@ func boundedSample(lower, upper interface{}) types.GomegaMatcher {
 		"Histogram": gstruct.Ignore(),
 	}))
 }
+
+func haveKeys(keys ...string) types.GomegaMatcher {
+	gomega.Expect(keys).ToNot(gomega.BeEmpty())
```
Comment (ffromani) [inline test/e2e_node/resource_metrics_test.go:178]: I'd use `ExpectWithOffset` - not sure if there's a better method mark this function as helper

File: test/e2e_node/resource_metrics_test.go [hunk 2]
```diff
@@ -73,7 +73,7 @@ var _ = SIGDescribe("ResourceMetricsAPI [NodeFeature:ResourceMetrics]", func() {
 			memoryCapacity := node.Status.Capacity["memory"]
 			memoryLimit := memoryCapacity.Value()
 
-			matchResourceMetrics := gstruct.MatchAllKeys(gstruct.Keys{
+			matchResourceMetrics := gomega.And(gstruct.MatchKeys(gstruct.IgnoreMissing, gstruct.Keys{
```
Comment (ffromani) [inline test/e2e_node/resource_metrics_test.go:76]: how hard is to detect if the swap is neabled in the node programmatically, and just use different `MatchAllKeys` in the two flows?

File: test/e2e_node/resource_metrics_test.go (hunk 1 above)
Comment (iholder101) [reply test/e2e_node/resource_metrics_test.go]: Thanks, done :+1:

File: test/e2e_node/resource_metrics_test.go (hunk 2 above)
Comment (ffromani) [reply test/e2e_node/resource_metrics_test.go:76]: I guess we can fix the failure and iterate later

Comment (iholder101) [reply test/e2e_node/resource_metrics_test.go:76]: I don't know of an easy way to do so.

Even is `NodeSwap` feature-gate is enabled, it doesn't mean that the node actually has swap provisioned on it. We could always try to find the node that the pod runs on and try executing stuff on it to find this out, but I think it's a big enough complication.

Comment (iholder101) [reply test/e2e_node/resource_metrics_test.go:76]: > I guess we can fix the failure and iterate later

Sounds good to me :+1: Thanks!

Comment (ffromani) [reply test/e2e_node/resource_metrics_test.go:76]: thinking aloud, maybe we can use `swapon`? Anyway, the more I think about it the more it seems wiser to fix the burning issue and improve later, the current approach seems good enough for the time being.

If you resubmit for whatever reason, please add copy this content https://github.com/kubernetes/kubernetes/pull/119421#issuecomment-1641892456 as code comment right above the `gomega.And`

**Review by ffromani — COMMENTED** [review]
/lgtm

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

**Comment by pacoxu** [comment]
https://prow.k8s.io/job-history/gs/kubernetes-jenkins/logs/ci-crio-cgroupv1-node-e2e-features
is green now.

test failures(7 today) [link](https://storage.googleapis.com/k8s-triage/index.html#6c0d381b56fe6946e061)

**Pull request merged**